examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +126,43 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+	pid_t child_pid;
+	int statusw;
+	int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+	if (fd < 0) { 
+		return false; 
+		}
+	switch (child_pid = fork()) 
+	{
+  	case -1: 
+  		return false;
+  	case 0:
+    		if (dup2(fd, 1) < 0) { 
+    			return false; 
+    			}
+    		if(execv(command[0],command)==-1)
+		exit(EXIT_FAILURE);
+  	default:
+    		/* do whatever the parent wants to do. */
+    		if(waitpid(child_pid,&statusw,0)==-1){
+			return false;
```

Comment:
@akshithaluguri22 cleanup missing (`close(fd`), `va_end(args)`) in all of these return paths.